### PR TITLE
Create GAN.py

### PR DESCRIPTION
**Pull request by DevilLord9967**
Adding GAN Neural Network Algorithm Implementation

### CODE REVIEW

File: neural_network/GAN.py [hunk 1]
```diff
@@ -1,9 +1,7 @@
-import numpy as np,sys,time
```
Comment (cclauss) [inline neural_network/GAN.py]: Let's just run `isort` on all the imports...
```
import sys

import matplotlib.gridspec as gridspec
import matplotlib.pyplot as plt
import numpy as np
from sklearn.utils import shuffle
from tensorflow.examples.tutorials.mnist import input_data
```

Comment (cclauss) [reply neural_network/GAN.py]: https://github.com/timothycrosley/isort

**Comment by cclauss** [comment]
Automated tests are failing... https://travis-ci.com/TheAlgorithms/Python/builds/133626860#L344

File: neural_network/GAN.py (hunk 1 above)
Comment (DevilLord9967) [reply neural_network/GAN.py]: I ran the isort.
Thanks for this wonderful information.

**Comment by DevilLord9967** [comment]
That input data import is still not working!!!
I gotta think another way around.

**Comment by cclauss** [comment]
https://chromium.googlesource.com/external/github.com/tensorflow/tensorflow/+/r0.7/tensorflow/g3doc/tutorials/mnist/beginners/index.md

**Comment by DevilLord9967** [comment]
@cclauss Is it okay to add the import like this or should i make another directory for that??

**Pull request merged**